nixos/doc/manual/release-notes/rl-2305.section.md

```diff
@@ -105,6 +105,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - Calling `makeSetupHook` without passing a `name` argument is deprecated.
 
+- `lib.systems.examples.ghcjs` and consequently `pkgsCross.ghcjs` now use the target triplet `javascript-unknown-ghcjs` instead of `js-unknown-ghcjs`. This has been done to match an [upstream decision](https://gitlab.haskell.org/ghc/ghc/-/commit/6636b670233522f01d002c9b97827d00289dbf5c) to follow Cabal's platform naming more closely. Nixpkgs will also reject `js` as an architecture name.
```

Comment:
Hrm, I thought `lib/systems` tracked `gnu-config` (and maybe-sometimes-LLVM).  Should we really be adding more sources of truth to the tracking-set?

Can't we just convert these tuples to Cabal-compatible format on the fly?  I call the conversion tables ["Rosetta Stones"](https://github.com/NixOS/nixpkgs/search?q=%22rosetta+stone%22&type=issues); we already have them for golang, openssl, linux, java and at least a dozen other large software environments that made up their own taxonomies instead of using autoconf's.

If upstream Cabal consciously decided to use tuples that `gnu-config` can't parse, then we should just translate our internal `gnu-config`-compatible representation into their representation at the point of use.

Reply:
BTW this isn't urgent, nor is it something I feel super strongly about.  I just figured if we were making a breaking change anyways now would be a good time to get back to "`config` should be accepted by `gnu-config`".

If you don't want to delay this PR with this sort of tuple-fiddling I can open a separate PR with my suggested change above.